src/ServiceControl.Audit.Persistence.RavenDb5/RavenDbPersistenceConfiguration.cs

```diff
@@ -11,25 +11,57 @@ public IPersistence Create(PersistenceSettings settings)
             var databaseConfiguration = GetDatabaseConfiguration(settings);
             var databaseSetup = new DatabaseSetup(databaseConfiguration);
 
-            return new RavenDb5Persistence(databaseConfiguration, databaseSetup, settings);
+            return new RavenDb5Persistence(databaseConfiguration, databaseSetup);
         }
-        static DatabaseConfiguration GetDatabaseConfiguration(PersistenceSettings settings)
+
+        internal static DatabaseConfiguration GetDatabaseConfiguration(PersistenceSettings settings)
         {
-            if (!settings.PersisterSpecificSettings.TryGetValue("ServiceControl/Audit/RavenDb5/DatabaseName", out var databaseName))
+            if (!settings.PersisterSpecificSettings.TryGetValue(DatabaseNameKey, out var databaseName))
             {
                 databaseName = "audit";
             }
 
+            ServerConfiguration serverConfiguration;
+
+            if (settings.PersisterSpecificSettings.TryGetValue(DatabasePathKey, out var dbPath))
+            {
+                if (settings.PersisterSpecificSettings.ContainsKey(ConnectionStringKey))
+                {
+                    throw new InvalidOperationException($"Both {DatabasePathKey} and {ConnectionStringKey} cant be specified at the same ftime");
+                }
+
+                var hostName = settings.PersisterSpecificSettings[HostNameKey];
+                var databaseMaintenancePort =
+                    int.Parse(settings.PersisterSpecificSettings[DatabaseMaintenancePortKey]);
```

Comment:
```suggestion
                var databaseMaintenancePort = int.Parse(settings.PersisterSpecificSettings[DatabaseMaintenancePortKey]);
```

Is a `TryParse` along with a meaningful exception better?

Reply:
Agreed, fixed and added tests